samples/attested_tls/server/enc/openssl_server.cpp

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <arpa/inet.h>
+#include <openenclave/enclave.h>
+#include <openssl/evp.h>
+#include <openssl/ssl.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <unistd.h>
+#include "../../common/openssl_utility.h"
+
+extern "C"
+{
+    int setup_tls_server(char* server_port, bool keep_server_up);
+};
+
+int verify_callback(int preverify_ok, X509_STORE_CTX* ctx);
+
+int initalize_ssl_context(SSL_CTX*& ctx)
+{
+    int ret = -1;
+    const SSL_METHOD* method;
+    if ((ctx = SSL_CTX_new(SSLv23_server_method())) == NULL)
+    {
+        printf(TLS_SERVER " unable to create a new SSL context\n");
+        goto exit;
+    }
+    // choose TLSv1.2 by excluding SSLv2, SSLv3 ,TLS 1.0 and TLS 1.1
+    SSL_CTX_set_options(ctx, SSL_OP_NO_SSLv2);
+    SSL_CTX_set_options(ctx, SSL_OP_NO_SSLv3);
+    SSL_CTX_set_options(ctx, SSL_OP_NO_TLSv1);
+    SSL_CTX_set_options(ctx, SSL_OP_NO_TLSv1_1);
+    ret = 0;
+exit:
+    return ret;
+}
+
+int create_listener_socket(int port, int& server_socket)
+{
+    int ret = -1;
+    struct sockaddr_in addr;
+    addr.sin_family = AF_INET;
```

Comment:
Why IPv4 only?  This is bad practice for samples.

Reply:
Created issue #3751 to identify all such instances in OE sample applications and fix them.